WordPressKitTests/PluginDirectoryTests.swift

```diff
@@ -343,4 +343,16 @@ class PluginDirectoryTests: XCTestCase {
 
         XCTAssertTrue(lhs == rhs)
     }
+
+    func testUnconventionalPluginSlug() async throws {
+        let data = try MockPluginDirectoryProvider.getPluginDirectoryMockData(with: "plugin-directory-rename-xml-rpc", sender: type(of: self))
+        stub(condition: isHost("api.wordpress.org")) { _ in
+            HTTPStubsResponse(data: data, statusCode: 200, headers: ["Content-Type": "application/json"])
+        }
+
+        let _ = try await PluginDirectoryServiceRemote().getPluginInformation(slug: "%-is-not-allowed")
+        let _ = try await PluginDirectoryServiceRemote().getPluginInformation(slug: "中文")
+
+        // No assertion needed.
+    }
```

Comment:
A suggestion for your consideration.

```suggestion
        await XCTAssertNoThrow(try await PluginDirectoryServiceRemote().getPluginInformation(slug: "%-is-not-allowed"))
        await XCTAssertNoThrow(try await PluginDirectoryServiceRemote().getPluginInformation(slug: "中文"))
    }
}

func XCTAssertNoThrow<T>(
    _ expression: @autoclosure () async throws -> T,
    _ message: @autoclosure () -> String = "",
    file: StaticString = #filePath,
    line: UInt = #line
) async {
    let error: Error?
    do {
        _ = try await expression()
        error = nil
    } catch let caughtError {
        error = caughtError
    }

    XCTAssertNil(
        error,
        message().isEmpty ? "Expected to not throw, threw \(String(describing: error))" : message(),
        file: file,
        line: line
    )
}
```

The motivation for this is that a test with no assertion looks odd to me. But, having said that, XCTest will fail a test `test name() throws` that threw an error, so it might be much of a muchness...

<img width="1967" alt="image" src="https://github.com/wordpress-mobile/WordPressKit-iOS/assets/1218433/cf5cf606-cab4-4bd2-b96c-f916ae9ad10b">

Reply:
> XCTest will fail a test test name() throws that threw an error, so it might be much of a muchness...

That's my thinking too. XCTest kind of makes the "no throw" assertion for us.